tests/abort4.rs

```diff
@@ -0,0 +1,180 @@
+                3,
+                Amount::from_btc(0.05).unwrap(),
+            )
+        })
+        .collect::<Vec<_>>();
+
+    // Fund the Maker with 4 utxos of 0.05 btc each and do basic checks on the balance.
+    let makers_ref = makers.iter().map(Arc::as_ref).collect::<Vec<_>>();
+    fund_and_verify_maker(
+        makers_ref,
+        &test_framework.bitcoind,
+        4,
+        Amount::from_btc(0.05).unwrap(),
+    );
+
+    // Start the Maker Server threads
+    log::info!("Initiating Maker...");
+    let maker_threads = makers
+        .iter()
+        .map(|maker| {
+            let maker_clone = maker.clone();
+            thread::spawn(move || {
+                start_maker_server(maker_clone).unwrap();
+            })
+        })
+        .collect::<Vec<_>>();
+
+    // Makers take time to fully setup.
+    let org_maker_spend_balances = makers
+        .iter()
+        .map(|maker| {
+            while !maker.is_setup_complete.load(Relaxed) {
+                log::info!("Waiting for maker setup completion");
+                // Introduce a delay of 10 seconds to prevent write lock starvation.
+                thread::sleep(Duration::from_secs(10));
+                continue;
+            }
+
+            // Check balance after setting up maker server.
+            let wallet = maker.wallet.read().unwrap();
+            let all_utxos = wallet.get_all_utxo().unwrap();
+
+            let balances = wallet.get_balances(Some(&all_utxos)).unwrap();
+
+            assert_eq!(balances.regular, Amount::from_btc(0.14999).unwrap());
+            assert_eq!(balances.fidelity, Amount::from_btc(0.05).unwrap());
+            assert_eq!(balances.swap, Amount::ZERO);
+            assert_eq!(balances.contract, Amount::ZERO);
+
+            balances.spendable
+        })
+        .collect::<Vec<_>>();
+
+    // Initiate Coinswap for both Takers concurrently
+    log::info!("Initiating coinswap protocol for multiple takers");
+
+    // Spawn threads for each taker to initiate coinswap concurrently
+    let mut takers_arc = takers
+        .into_iter()
+        .map(|t| Arc::new(Mutex::new(t)))
+        .collect::<Vec<_>>();
+
+    let taker_threads = takers_arc
+        .iter_mut()
+        .map(|taker| {
+            let taker_clone = taker.clone();
+            thread::spawn(move || {
+                // Swap params for coinswap
+                let swap_params = SwapParams {
+                    send_amount: Amount::from_sat(500000),
+                    maker_count: 2,
+                    tx_count: 3,
+                    required_confirms: 1,
+                };
+                taker_clone
+                    .lock()
+                    .unwrap()
+                    .do_coinswap(swap_params)
+                    .unwrap();
+            })
+        })
+        .collect::<Vec<_>>();
+
+    // Wait for taker threads to complete
+    for thread in taker_threads {
+        let _ = thread.join();
+    }
```

Comment:
We can instead use scoped threads...
```suggestion
thread::scope(|s| {
        for taker in &mut takers {
            s.spawn( move || {
                taker.do_coinswap(swap_params).unwrap();
            });
        }
    });
```